azurerm/resource_arm_app_service_certificate.go

```diff
@@ -0,0 +1,250 @@
+	}
+
+	if v, ok := d.GetOk("pfx_blob"); ok {
+		pfxBlob, err := base64.StdEncoding.DecodeString(v.(string))
+		if err != nil {
+			return fmt.Errorf("Could not decode PFX blob: %+v", err)
+		}
+		certificate.CertificateProperties.PfxBlob = &pfxBlob
+	}
+
+	if v, ok := d.GetOk("key_vault_id"); ok {
+		certificate.CertificateProperties.KeyVaultID = utils.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("key_vault_secret_name"); ok {
+		certificate.CertificateProperties.KeyVaultSecretName = utils.String(v.(string))
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroup, name, certificate); err != nil {
+		return err
+	}
+
+	read, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read App Service Certificate %q (Resource Group %q) ID", name, resourceGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmAppServiceCertificateRead(d, meta)
+}
+
+func resourceArmAppServiceCertificateRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).web.CertificatesClient
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	name := id.Path["certificates"]
+
+	ctx := meta.(*ArmClient).StopContext
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] App Service Certificate %q (Resource Group %q) was not found - removing from state", name, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on App Service Certificate %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if props := resp.CertificateProperties; props != nil {
+		d.Set("friendly_name", props.FriendlyName)
+		d.Set("subject_name", props.SubjectName)
+		d.Set("host_names", props.HostNames)
+		d.Set("issuer", props.Issuer)
+		d.Set("issue_date", props.IssueDate)
+		d.Set("expiration_date", props.ExpirationDate)
+		d.Set("thumbprint", props.Thumbprint)
+		d.Set("key_vault_id", props.KeyVaultID)
+		d.Set("key_vault_secret_name", props.KeyVaultSecretName)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmAppServiceCertificateDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).web.CertificatesClient
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["certificates"]
+
+	log.Printf("[DEBUG] Deleting App Service Certificate %q (Resource Group %q)", name, resourceGroup)
+
+	ctx := meta.(*ArmClient).StopContext
+	resp, err := client.Delete(ctx, resourceGroup, name)
+	if err != nil {
+		if !utils.ResponseWasNotFound(resp) {
+			return err
```

Comment:
can we wrap this error:

```suggestion
			return fmt.Errorf("Error deleting App Service Certificate %q (Resource Group %q): %s)", name, resourceGroup, err)
```